ethereum/client.go

```diff
@@ -210,6 +210,87 @@ func toBlockNumArg(number *big.Int) string {
 	return hexutil.EncodeBig(number)
 }
 
+// Transaction returns the transaction response of the Transaction identified
+// by *RosettaTypes.TransactionIdentifier hash
+func (ec *Client) Transaction(
+	ctx context.Context,
+	blockIdentifier *RosettaTypes.BlockIdentifier,
+	transactionIdentifier *RosettaTypes.TransactionIdentifier,
+) (*RosettaTypes.Transaction, error) {
+	if transactionIdentifier.Hash != "" {
+		var raw json.RawMessage
+		err := ec.c.CallContext(ctx, &raw, "eth_getTransactionByHash", transactionIdentifier.Hash)
+		if err != nil {
+			return nil, fmt.Errorf("%w: transaction fetch failed", err)
+		} else if len(raw) == 0 {
+			return nil, ethereum.NotFound
+		}
+
+		// Decode transaction
+		var body rpcTransaction
+
+		if err := json.Unmarshal(raw, &body); err != nil {
+			return nil, err
+		}
+
+		var header *types.Header
+		if blockIdentifier.Hash != "" {
+			header, err = ec.blockHeaderByHash(ctx, blockIdentifier.Hash)
+		} else {
+			header, err = ec.blockHeaderByNumber(ctx, big.NewInt(blockIdentifier.Index))
+		}
+
+		if err != nil {
+			return nil, fmt.Errorf("%w: could not get block header for %x", err, blockIdentifier.Hash)
+		}
+
+		receipt, err := ec.transactionReceipt(ctx, body.tx.Hash())
+		if receipt.BlockHash != *body.BlockHash {
```

Comment:
Do we really need this check? I feel ethclient in go-ethereum lib must validate this.

Reply:
`transactionReceipt` for tx receipt we just pass txhash so this function won't validate this. So we want to make sure that block hash matches the block hash in receipt by ourselves